### PR TITLE
[FLOC-1255] Convergence loop design sketch

### PR DESCRIPTION
**Pull request by itamarst**
https://clusterhq.atlassian.net/browse/FLOC-1255

As a first pass, just explain the state machine. I think this covers the necessary edge cases; if it's hard to understand I can turn it into more concrete sketched code... and regardless I may end up with second design review pass, we'll see.

### CODE REVIEW

File: flocker/node/_loop.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+If connection failed try to connect again.
+
+IGNORANT:
+
+The status of the cluster is unknown.
+If desired configuration and cluster state are received send a GO input
+symbol to the AgentOperation state machine. switch to KNOWN.
+If disconnected then switch to DISCONNECTED.
+
+KNOWLEDGEABLE:
```
Comment (exarkun) [inline flocker/node/_loop.py]: KNOWN / KNOWLEDGEABLE are the same I guess

**Comment by exarkun** [comment]
What functionality will be provided when this is complete?

**Comment by itamarst** [comment]
Given an AMP protocl implementation, a `flocker-p2p-agent` script that runs as a convergence agent, I guess.

**Comment by exarkun** [comment]
That can actually converge a node using our current convergence loop implementation?  Okay cool.

**Pull request closed without merging**